jerry-core/ecma/base/ecma-helpers-number.c

```diff
@@ -718,6 +718,32 @@ ecma_number_multiply (ecma_number_t left_num, /**< left operand */
   return left_num * right_num;
 } /* ecma_number_multiply */
 
+/**
+ * ECMA-integer number multiplication.
+ *
+ * @return number - result of multiplication.
+ */
+inline ecma_value_t __attr_always_inline___
+ecma_integer_multiply (ecma_integer_value_t left_integer, /**< left operand */
+                       ecma_integer_value_t right_integer) /**< right operand */
+{
+#ifdef __GNUC__
```

Comment:
There is no need to be gcc specific.

Reply:
It should be. Also clang specified as well.

Reply:
@zherczeg I think there are already several gcc and clang specific code in jerry. For example, [`__attr_always_inline___`](https://github.com/jerryscript-project/jerryscript/blob/master/jerry-core/jrt/jrt.h#L33) uses gcc and clang specific attribute. I am suspecting `__attr_const__` and `__attr_pure___` are similar cases. I think we should enclose these definitions with `#if defined(__clang__) || defined(__GNUC__)`.